autoscale/src/main/java/com/sequenceiq/periscope/monitor/handler/UpdateFailedHandler.java

```diff
@@ -12,21 +14,29 @@
 import org.springframework.stereotype.Component;
 
 import com.sequenceiq.cloudbreak.logger.MDCBuilder;
-import com.sequenceiq.periscope.api.model.ClusterState;
+import com.sequenceiq.cloudbreak.message.CloudbreakMessagesService;
+import com.sequenceiq.periscope.api.model.ScalingStatus;
 import com.sequenceiq.periscope.domain.Cluster;
 import com.sequenceiq.periscope.monitor.event.UpdateFailedEvent;
 import com.sequenceiq.periscope.service.ClusterService;
+import com.sequenceiq.periscope.service.HistoryService;
 
 @Component
 public class UpdateFailedHandler implements ApplicationListener<UpdateFailedEvent> {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(UpdateFailedHandler.class);
 
-    private static final int RETRY_THRESHOLD = 5;
+    private static final int RETRY_THRESHOLD = 15;
 
     @Inject
     private ClusterService clusterService;
 
+    @Inject
+    private HistoryService historyService;
+
+    @Inject
+    private CloudbreakMessagesService messagesService;
+
     private final Map<Long, Integer> updateFailures = new ConcurrentHashMap<>();
```

Comment:
Are failures ever reset?, other than at the point when the cluster is being disabled (or Periscope restarts)?

Reply:
It is reset as soon as AS is suspended.